arch/arm/core/cortex_m/mpu/arm_mpu.c

```diff
@@ -64,7 +64,7 @@ static void _region_init(u32_t index, u32_t region_addr,
 	/* Select the region you want to access */
 	ARM_MPU_DEV->rnr = index;
 	/* Configure the region */
-	ARM_MPU_DEV->rbar = region_addr | REGION_VALID | index;
+	ARM_MPU_DEV->rbar = region_addr & REGION_ADDR_MASK;
```

Comment:
Thanks for pointing this out :smile: 

Even if I agree with the solution it does not seem to me to be related to the "mpu_stack_guard_test" example.

The issue here seems that if the last 5 bits of the RBAR field are set using only logic OR without masking them before might result in an unwanted behavior.

Masking them we are extending the stack region, but the stack_guard reserved region is big enough to prevent problems.

Said that I think we should keep the "REGION_VALID | index" information because it can be used in future for example to validate the operations we are doing on the MPU.

Could you please change this line as per below?

ARM_MPU_DEV->rbar = (region_addr & REGION_ADDR_MASK) | REGION_VALID | index;

Thanks